src/zope/tales/tests/test_expressions.py

```diff
@@ -261,15 +316,49 @@ def check(expr):
 
     def test_defer_expression_returns_wrapper(self):
         from zope.tales.expressions import DeferWrapper
+        from zope.tales.expressions import DeferExpr
+        expr = self.engine.compile('defer: b')
+        self.assertIsInstance(expr, DeferExpr)
+        self.assertEqual(str(expr), "<DeferExpr 'b'>")
+        self._check_evals_to_instance(expr, DeferWrapper)
+
+        wrapper = expr(self.context)
+        self.assertEqual(wrapper(), self.context.vars['b'])
+
+        self.assertEqual(str(wrapper), 'boot')
```

Comment:
AFAICS `boot` is the value of `self.context.vars['b']`.  It's unclear to me, while I'm looking at this test, why the expected value is spelled differently in the two assertions.

It's also unclear to me whether `wrapper()` and `str(wrapper)` are supposed to return the same thing always, or if perhaps `str(wrapper)` should be equivalent to `str(wrapper())`?  Maybe there's a context variable that holds a non-string value that could be used here?

Reply:
It's the latter case: `str(wrapper)` is meant to be the same as `str(wrapper())`. I thought that by spelling it equal to the context variable in the first case I was emphasizing that fact, but I suppose I wasn't. I'll see if I can make it more clear.